File: 1679-max-number-of-k-sum-pairs/main.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        
        count_pairs = 0
        count_map = {}

        for num in nums:
            complement = k - num
            if count_map.get(complement, 0) > 0:
                count_pairs += 1
                count_map[complement] -= 1
            else:
                count_map[num] = count_map.get(num, 0) + 1
        
        return count_pairs

File: 1679-max-number-of-k-sum-pairs/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_operations(nums: Vec<i32>, k: i32) -> i32 {
        
        let mut count_map: HashMap<i32, i32> = HashMap::new();
        let mut count_pairs = 0;


        for &num in &nums {
            let complement = k - num;
            match count_map.get_mut(&complement) {
                Some(count) if *count > 0 => {
                    count_pairs += 1;
                    *count -= 1;
                },
                _ => {
                    *count_map.entry(num).or_insert(0) += 1;
                }
            }
        }

        count_pairs
    }
}
